File: M/taskM.rb
sum = gets.chomp.to_i
num_1 = [1]
num_2 = [2]
while num_1.sum < sum
  num_1.first == 1 ? num_1.unshift(2) : num_1.unshift(1)
end

while num_2.sum < sum
  num_2.first == 1 ? num_2.unshift(2) : num_2.unshift(1)
end

if num_1.sum == sum && num_2.sum == sum
  num_1.join > num_2.join ? puts(num_1.join) : puts(num_2.join)
elsif num_1.sum == sum
  puts num_1.join
else
  puts num_2.join
end

File: M/taskM.py
input_sum = int(input())
num_1 = [1]
num_2 = [2]

while sum(num_1) < input_sum:
  if num_1[0] == 1:
    num_1 = [2] + num_1 
  else:
    num_1 = [1] + num_1

while sum(num_2) < input_sum:
  if num_2[0] == 1:
    num_2 = [2] + num_2
  else:
    num_2 = [1] + num_2

if sum(num_1) == input_sum and sum(num_2) == input_sum:
  if int(''.join(str(n) for n in num_1)) > int(''.join(str(n) for n in num_2)):
    print(''.join(str(n) for n in num_1))
  else:
    print(''.join(str(n) for n in num_2))
elif sum(num_1) == input_sum:
  print(''.join(str(n) for n in num_1))
else:
  print(''.join(str(n) for n in num_2))
